frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -1118,10 +1132,15 @@ void OBSBasic::AddQuickTransitionId(int id)
 
 void OBSBasic::AddQuickTransition()
 {
-	int trIdx = sender()->property("transition_index").toInt();
+	std::string transitionUuid = sender()->property("transition_uuid").toString().toStdString();
 	QSpinBox *duration = sender()->property("duration").value<QSpinBox *>();
 	bool fadeToBlack = sender()->property("fadeToBlack").value<bool>();
-	OBSSource transition = fadeToBlack ? OBSSource(fadeTransition) : GetTransitionComboItem(ui->transitions, trIdx);
+	OBSSource transition;
+
+	if (!fadeToBlack && (transitions.find(transitionUuid) == transitions.end()))
+		return;
+
+	transition = fadeToBlack ? OBSSource(fadeTransition) : transitions[transitionUuid];
```

Comment:
Be aware that `transitions[transitionUuid]` can potentially an empty element with the transition UUID if none was found. If this is not desired, either use the getter methods introduced early or use `at` with exception handling.

As the `find` method is used already, it would be better to store the iterator and use `second` on that iterator to pass on the element instead of iterating over the collection again.

Reply:
Mutators removed